test/sql/updates/post.repair.sql

```diff
@@ -2,9 +2,26 @@
 -- Please see the included NOTICE for copyright information and
 -- LICENSE-APACHE for a copy of the license.
 
--- Re-add the dropped foreign key constraint that was dropped for
--- repair testing.
-ALTER TABLE _timescaledb_catalog.chunk_constraint
+SELECT extversion < '2.10.0' AS test_repair_dimension
+FROM pg_extension
+WHERE extname = 'timescaledb' \gset
+
+SELECT extversion >= '2.10.0' AS has_cagg_joins
+FROM pg_extension
+WHERE extname = 'timescaledb' \gset
+
+SELECT :'TEST_VERSION' >= 'v8' AS is_GTE_v8 \gset
+
+\if :test_repair_dimension
+    -- Re-add the dropped foreign key constraint that was dropped for
+    -- repair testing.
+    ALTER TABLE _timescaledb_catalog.chunk_constraint
     ADD CONSTRAINT chunk_constraint_dimension_slice_id_fkey
     FOREIGN KEY (dimension_slice_id) REFERENCES _timescaledb_catalog.dimension_slice (id);
+\endif
+
+\if :has_cagg_joins AND :is_GTE_v8
+--Check if the repaired cagg with joins work alright now
+\ir post.repair.cagg_joins.sql
+\endif
```

Comment:
So far there are no multiple files for the post.repair, since we are not adding not another version as part of this, I am not sure if this required as part of this PR.

Reply:
@mkindahl currently in `repair` tests we do not have the behavior to deal with multiple version of SQL test files... We decided to put the logic inside the SQL files to don't mess to much with the shell scripts in this PR because we're already struggling too much with this.

Reply:
Ok. Sounds good. Not critical.